src/Microsoft.AspNet.Hosting/IWebApplication.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Http.Features;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNet.Hosting
+{
+    /// <summary>
+    /// Represents a configured web application
+    /// </summary>
+    public interface IWebApplication
+    {
+        /// <summary>
+        /// The <see cref="IFeatureCollection"/> exposed by the configured server.
+        /// </summary>
+        IFeatureCollection ServerFeatures { get; }
+
+        /// <summary>
+        /// The <see cref="IServiceProvider"/> for the application.
+        /// </summary>
+        IServiceProvider Services { get; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        IApplicationLifetime Lifetime { get; }
```

Comment:
They are services that are created by the Builder but you're right, it might be weird. But they are so commonly used.